new_sfx.py

```diff
@@ -0,0 +1,272 @@
+
+
+LOG: logging.Logger             = logging.getLogger(f"[{Path(__file__).name}]")
+LOG_LEVEL: int                  = logging.INFO
+
+EXTRACTED_PATH: Path            = Path("extracted")
+SAMPLEBANK_NAME: str            = "SampleBank_0"
+ASSETS_SAMPLES_PATH: Path       = Path("assets") / "audio" / "samples" / SAMPLEBANK_NAME 
+ASSETS_SAMPLEBANK_XML: Path     = Path("assets") / "audio" / "samplebanks" / (SAMPLEBANK_NAME + ".xml")
+ASSETS_SOUNDFONT_XML: Path      = Path("assets") / "audio" / "soundfonts" / "Soundfont_0.xml"
+ASSETS_SEQUENCE_SEQ: Path       = Path("assets") / "audio" / "sequences" / "seq_0.prg.seq"
+REPLACEMENT_PATTERN: str        = "[^0-9a-zA-Z]+"
+SFX_BANK_HEADER: Path           = Path("include") / "tables" / "sfx" / "environmentbank_table.h"
+DEFINE_FORMATSTR: str           = "             DEFINE_SFX({0}, {1}, 0x80, 0, 0, 0)"
+
+
+class FFMPEGOptions:
+    def __init__(self, volume:str=None, sample_rate:str="32000"):
+        self.volume:str = volume
+        self.sample_rate:str = sample_rate
+
+class CustomSoundEffect:
+    def __init__(self, sample_path: Path):
+        self.source_path: Path = sample_path
+        self.dest_path: Path = CustomSoundEffect._safe_rename(sample_path)
+        id_base = self.dest_path.stem.upper()
+        self.sample_identifier: str     = f"CUSTOM_SAMPLE_{id_base}"
+        self.effect_identifier: str     = f"CUSTOM_EFFECT_{id_base}"
+        self.channel_identifier: str    = f"CHAN_CUSTOM_{id_base}"
+        self.layer_identifier: str      = f"LAYER_CUSTOM_{id_base}"
+        self.enum_identifier: str       = f"NA_SE_EV_CUSTOM_{id_base}"
+    
+    def _safe_rename(sample_path: Path) -> Path:
+        new_name: str = re.sub(REPLACEMENT_PATTERN, "_", sample_path.stem) + sample_path.suffix
+        if new_name != sample_path.name:
+            LOG.info(f"sanitizing name of sample, new name: '{new_name}'")
+        
+        new_path = ASSETS_SAMPLES_PATH / new_name 
+        if new_path.exists():
+            LOG.warning(f"filename already exists! sample will be overwritten! '{new_name}'")
+
+        return new_path
+
+
+def _write_xml_pretty(tree: xml.etree.ElementTree, path: Path) -> None:
+    rough_string = xml.etree.ElementTree.tostring(tree, 'utf-8')
+    reparsed = xml.dom.minidom.parseString(rough_string)
+    pretty = reparsed.toprettyxml(indent="    ")
+    trimmed = os.linesep.join([s for s in pretty.splitlines() if s.strip()])
+    with open(path, "w") as fd:
+        fd.write(trimmed)
+        fd.write("\n")
+
+
+def prepare_directories() -> None:
+    os.makedirs(ASSETS_SAMPLES_PATH, exist_ok=True)
+    os.makedirs(ASSETS_SAMPLEBANK_XML.parent, exist_ok=True)
+    os.makedirs(ASSETS_SOUNDFONT_XML.parent, exist_ok=True)
+
+    games: list[str] = os.listdir(EXTRACTED_PATH)
+    if len(games) == 0:
+        LOG.error("No assets found in extracted/, 'make setup' not yet run?")
+        sys.exit(1)
+    # just use the first one found, it's fine (probably)
+    game = games[0]
```

Comment:
this is what i am referring to in my previous comment above. prolly wanna edit this part to be hackeroot-y

Reply:
yeah I think having a `--version -v` argument would be convenient for HackerOoT, people don't necessarily use ntsc-1.2 (which is the default currently)

Reply:
right. Oh or probably reading it from the Makefile would be better

Reply:
I think both would be neat actually, that said I don't think anyone would use several versions at the same time lol